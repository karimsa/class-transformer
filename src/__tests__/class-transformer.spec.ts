import { test, expect } from '@jest/globals'

import { Type, plainToClass } from '../'

const expectType = <T>(value: T) => value

test('should parse dates correctly', () => {
	class TestClass {
		@Type(() => Date)
		date: Date
	}

	expect(
		expectType<TestClass>(
			plainToClass(TestClass, {
				date: new Date().toISOString(),
			})
		)
	).toMatchSnapshot({
		date: expect.any(Date),
	})
})
