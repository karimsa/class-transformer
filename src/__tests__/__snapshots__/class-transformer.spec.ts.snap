// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse dates correctly 1`] = `
Object {
  "date": Any<Date>,
}
`;
